File: Belajar BWA/HandleAPI/app/src/main/java/com/example/handleapi/MainActivity.java
package com.example.handleapi;

import androidx.appcompat.app.AppCompatActivity;

import android.app.ProgressDialog;
import android.os.AsyncTask;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.TextView;

import org.json.JSONArray;
import org.json.JSONException;
import org.json.JSONObject;

import java.io.InputStream;
import java.io.InputStreamReader;
import java.net.HttpURLConnection;
import java.net.URL;

import javax.net.ssl.HttpsURLConnection;

public class MainActivity extends AppCompatActivity {

    private ProgressDialog progressDialog;
    private TextView tvShow;
    private Button btnRefresh;

    private String apiUrl = "https://61fb4b3887801d0017a2c45c.mockapi.io/produk";

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        initView();
        initListener();
        getDataHome();
    }

    private void getDataHome() {
        getHomeAsyncTask getHomeAsyncTask = new getHomeAsyncTask();
        getHomeAsyncTask.execute();
    }

    private void initListener() {
        btnRefresh.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                getDataHome();
            }
        });

    }

    private void initView() {
        tvShow = findViewById(R.id.tv_show);
        btnRefresh = findViewById(R.id.btn_refresh);
    }
     public class getHomeAsyncTask extends AsyncTask<String, String, String> {
        @Override
        protected void onPreExecute(){
            super.onPreExecute();

            progressDialog = new ProgressDialog(MainActivity.this);
            progressDialog.setMessage("Please wait...");
            progressDialog.setCancelable(false);
            progressDialog.show();
        }

        @Override
        protected String doInBackground(String... strings){
             String current ="";
             try{
                 URL url;
                 HttpURLConnection urlConnection = null;
                 try{
                     url = new URL(apiUrl);
                     urlConnection = (HttpsURLConnection) url.openConnection();

                     InputStream in = urlConnection.getInputStream();
                     InputStreamReader isw = new InputStreamReader(in);

                     int data = isw.read();
                     while(data != -1){
                         current += (char) data;
                         data = isw.read();
                     }
                     return current;
                 }catch(Exception e){
                     e.printStackTrace();
                 }finally {
                     if (urlConnection != null){
                         urlConnection.disconnect();
                     }
                 }
             }catch(Exception e){
                 e.printStackTrace();
                 return "Exception : " + e.getMessage();
             }
             return current;
        }

        @Override
        protected void onPostExecute(String s){
            progressDialog.dismiss();

            try{
                JSONArray jsonArray = new JSONArray(s);
                JSONObject object = jsonArray.getJSONObject(0);
                String showData = "Nama Produk : "+ object.getString("nama_produk");

                tvShow.setText(showData);
            }catch (JSONException e){
                e.printStackTrace();
            }
        }
     }

}

File: Calculator/app/src/main/java/com/example/calculator/MainActivity.kt
package com.example.calculator

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Button
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    private var canAddOperation = false
    private var canAddDecimal = true

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    fun numberAction(view: View){
        if (view is Button){
            if(view.text == "."){
                if(canAddDecimal)
                    workingsTV.append(view.text)
                canAddOperation = false
            }
            else
                workingsTV.append(view.text)

            canAddOperation = true
        }
    }

    fun operationAction(view: View){
        if (view is Button && canAddOperation){
            workingsTV.append(view.text)
            canAddOperation = false
            canAddDecimal = true
        }
    }

    fun allClearAction(view: View)
    {
        workingsTV.text=""
        resultsTV.text=""
    }
    fun backSpaceAction(view: View)
    {
        val length = workingsTV.length()
        if (length > 0)
            workingsTV.text = workingsTV.text.subSequence(0, length - 1)
    }
    fun equalsAction(view: View) {}
}